openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/BallerinaServiceGenerator.java

```diff
@@ -268,7 +269,8 @@ private FunctionDefinitionNode getResourceFunction(Map.Entry<PathItem.HttpMethod
             isNullableRequired = parametersGenerator.isNullableRequired();
         }
         SeparatedNodeList<ParameterNode> parameters = createSeparatedNodeList(params);
-        String pathForRecord = Objects.equals(path, SLASH) ? "" : GeneratorUtils.getValidName(path, true);
+        String pathForRecord = Objects.equals(path, SLASH) || Objects.equals(path, CATCH_ALL_PATH) ? "" :
+                GeneratorUtils.getValidName(path, true);
```

Comment:
Any specific reason to use `Objects.equals()` in here? Can't we just do `str1.equals(str2)` instead?